src/jaxsim/rbda/contacts/visco_elastic.py

```diff
@@ -0,0 +1,1050 @@
+        ):
+            W_ν̇_pr = jnp.hstack(
+                js.ode.system_acceleration(
+                    model=model,
+                    data=data,
+                    joint_force_references=references.joint_force_references(
+                        model=model
+                    ),
+                    link_forces=W_f̅_L + references.link_forces(model=model, data=data),
+                )
+            )
+
+        # Compute the system mixed acceleration, used to integrate the system position.
+        # It considers the average of the average contact forces computed with the
+        # exponential integrator.
+        with (
+            data.switch_velocity_representation(jaxsim.VelRepr.Mixed),
+            references.switch_velocity_representation(jaxsim.VelRepr.Mixed),
+        ):
+            BW_ν̇_pr2 = jnp.hstack(
+                js.ode.system_acceleration(
+                    model=model,
+                    data=data,
+                    joint_force_references=references.joint_force_references(
+                        model=model
+                    ),
+                    link_forces=LW_f̿_L + references.link_forces(model=model, data=data),
+                )
+            )
+
+        # Integrate the system velocity using the inertial-fixed acceleration.
+        W_ν_plus = W_ν_t0 + dt * W_ν̇_pr
+
+        # Integrate the system position using the mixed velocity.
+        q_plus = jnp.hstack(
+            [
+                # Note: here both ṗ and p̈ -> need mixed representation.
+                W_p_B_t0 + dt * W_ṗ_B_t0 + 0.5 * dt**2 * BW_ν̇_pr2[0:3],
+                jaxsim.math.Quaternion.integration(
+                    dt=dt,
+                    quaternion=W_Q_B_t0,
+                    omega=(W_ω_WB_t0 + 0.5 * dt * BW_ν̇_pr2[3:6]),
+                    omega_in_body_fixed=False,
+                ).squeeze(),
+                s_t0 + dt * ṡ_t0 + 0.5 * dt**2 * BW_ν̇_pr2[6:],
+            ]
+        )
+
+        # Create the data at the final time.
+        with data.editable(validate=True) as data_tf:
+            data_tf: js.data.JaxSimModelData
+            data_tf.time_ns = data.time_ns + (dt * 1e9).astype(data.time_ns.dtype)
+
+        data_tf = data_tf.reset_joint_positions(q_plus[7:])
+        data_tf = data_tf.reset_base_position(q_plus[0:3])
+        data_tf = data_tf.reset_base_quaternion(q_plus[3:7])
+
+        data_tf = data_tf.reset_joint_velocities(W_ν_plus[6:])
+        data_tf = data_tf.reset_base_velocity(
+            W_ν_plus[0:6], velocity_representation=jaxsim.VelRepr.Inertial
+        )
+
+        return data_tf.replace(
+            velocity_representation=data.velocity_representation, validate=False
+        )
+
+
+@jax.jit
+def step(
+    model: js.model.JaxSimModel,
+    data: js.data.JaxSimModelData,
+    *,
+    dt: jtp.FloatLike,
+    link_forces: jtp.MatrixLike | None = None,
+    joint_force_references: jtp.VectorLike | None = None,
```

Comment:
`step` is called only internally right? So if I'm not wrong these two arguments should already be `jnp.array`
```suggestion
    link_forces: jtp.Matrix | None = None,
    joint_force_references: jtp.Vector | None = None,
```

Reply:
No this is a public API. It is a replacement of `jaxsim.api.model.step` compatible with all the features that visco-elastic model provide.